### PR TITLE
feat(query): improve like function

### PR DESCRIPTION
**Pull request by Dousir9**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR
Improve like function, when we can't use `OrdinalStr`, `StartOfPercent`, `EndOfPercent` and `SurroundByPercent`, if the like expression only includes `%`, e.g. `'A%r%w'`, we will use `SimplePattern` to get better performance: TPC-H(SF=100) Q13: 8.6s -> 6.3s.

```sql
From TPC-H Q13:
select count(*) from orders where o_comment not like '%pending%deposits%';

TPC-H(SF=100): 4.03s -> 1.61s
```
ci-benchmark:
截屏2023-07-05 16 17 32

- Closes #issue

### CODE REVIEW

**Comment by TCeason** [comment]
1. Has some invalid commit info. Better to reduce them
2. Could you provide  the performance  improvement

**Review by TCeason — APPROVED** [review]

**Comment by BohuTANG** [comment]
We should add some unit tests or logic tests to make sure this PR works as expected and not broken by others in the future.

File: src/query/functions/src/scalars/comparison.rs [hunk 1]
```diff
@@ -776,3 +802,178 @@ fn like(haystack: &[u8], pattern: &[u8]) -> bool {
...
+        }
+        segment_idx += 1;
+    }
+    true
+}
+
+#[test]
+fn test_check_pattern_type() {
+    // OrdinalStr
+    let pattern = "databend".as_bytes();
```
Comment (sundy-li) [inline src/query/functions/src/scalars/comparison.rs:911]:
```
let test_cases = vec![
    ("databend",  PatternType::OrdinalStr),
 .....
];

for (a, b) in test_cases {
  assert_eq!(
        b,
        check_pattern_type(a, false)
    );
}
```

**Comment by sundy-li** [comment]
Do you have any stateless test cases which cover `SimplePattern ` pattern?

**Comment by Dousir9** [comment]
> SimplePattern

Let me add some tests.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**